keras_cv/models/resnet_v2_backbone.py

```diff
@@ -482,189 +377,20 @@ def get_config(self):
             "trainable": self.trainable,
         }
 
-    @classmethod
-    def from_config(cls, config):
-        return cls(**config)
-
-
-def ResNet18V2(
-    *,
-    include_rescaling,
-    include_top,
-    classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classifier_activation="softmax",
-    name="resnet18",
-    **kwargs,
-):
-    """Instantiates the ResNet18 architecture."""
-
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet18V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet18V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet18V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=weights,
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="basic_block",
-        **kwargs,
-    )
-
-
-def ResNet34V2(
-    *,
-    include_rescaling,
-    include_top,
-    classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classifier_activation="softmax",
-    name="resnet34",
-    **kwargs,
-):
-    """Instantiates the ResNet34 architecture."""
-
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet34V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet34V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet34V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=weights,
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="basic_block",
-        **kwargs,
-    )
-
-
-def ResNet50V2(
-    *,
-    include_rescaling,
-    include_top,
-    classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classifier_activation="softmax",
-    name="resnet50v2",
-    **kwargs,
-):
-    """Instantiates the ResNet50V2 architecture."""
-
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet50V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet50V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet50V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=parse_weights(weights, include_top, "resnet50v2"),
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="block",
-        **kwargs,
-    )
-
-
-def ResNet101V2(
```

Comment:
a meta comment: but I do think that `ResNet50v2(...)` does read a bit nicer than `ResNetv2Backbone.from_pretrained(...)`

Reply:
We discussed this earlier -- I think we can add nice aliases like `class ResNet50V2Backbone(**kwargs)` that just call `ResNetv2Backbone.from_preset("resnet50_v2", **kwargs)`.

`ResNet50V2Backbone(weights="imagenet", **kwargs)` would call `ResNetv2Backbone.from_preset("resnet50_v2_imagenet", **kwargs)`.

This is a pure alias so there is no added maintenance / complexity or even change in the user-facing mental model.

Reply:
This isn't quite true because we also have to keep the `weights` strings in sync with the `presets`.

However, my major reservation is allowing multiple ways to do the same thing often adds more confusion it removes. Why did we create a new library only to be tied to 2016 conventions from `keras.applications`? We have a "backward compatible" library already---the original one!

Reply:
What about `include_top`? Would we be grafting this back on?

Reply:
Also: what design pattern would we be highlighting in the documentation?

Reply:
> What about include_top? Would we be grafting this back on?

For this I'd say not worth it.  In Docs I think it depends how the model is being used.

Reply:
I tried to add a readable, minimalist version of this without a lot of automation. Let me know what you think.